ruff_lsp/server.py

```diff
@@ -397,6 +408,35 @@ async def hover(params: HoverParams) -> Hover | None:
     return None
 
 
+@LSP_SERVER.feature(TEXT_DOCUMENT_DOCUMENT_LINK)
+async def document_link(params: DocumentLinkParams) -> list[DocumentLink] | None:
+    """LSP handler for textDocument/documentLink request."""
+    document = LSP_SERVER.workspace.get_document(params.text_document.uri)
+
+    links: list[DocumentLink] = []
+    for index, line in enumerate(document.lines):
+        for code, start, end in _parse_codes(line):
+            result = await _run_subcommand_on_document(
+                document,
+                VERSION_REQUIREMENT_LINTER,
+                args=["--explain", code, "--format", "json"],
+            )
+            if result.stdout:
+                name = json.loads(result.stdout.decode("utf-8")).get("name")
+                links.append(
+                    DocumentLink(
+                        range=Range(
+                            start=Position(line=index, character=start),
+                            end=Position(line=index, character=end),
+                        ),
+                        tooltip=f"Jump to {code} documentation",
+                        target=f"https://docs.astral.sh/ruff/rules/{name}",
+                    )
+                )
```

Comment:
Does this iterate over the entire document, checking every line?

Reply:
Yes, it does...